crates/node/lang/src/pnpm/dependency_path.rs

```diff
@@ -32,54 +18,71 @@ impl PnpmDependencyPath {
         !path.starts_with('/')
     }
 
-    pub fn parse(path: &str) -> Result<Self, DependencyPathError> {
+    pub fn parse(path: &str) -> Self {
         let is_absolute = Self::is_absolute(path);
         let mut parts = path.split('/').map(String::from).collect::<Vec<_>>();
+
         if !is_absolute {
             parts.remove(0);
         }
+
         let host = if is_absolute {
             Some(parts.remove(0))
         } else {
             None
         };
+
+        if parts.is_empty() {
+            return PnpmDependencyPath {
+                host,
+                is_absolute,
+                ..PnpmDependencyPath::default()
+            };
+        }
+
         let name = if parts[0].starts_with('@') {
             Some(format!("{}/{}", parts.remove(0), parts.remove(0)))
         } else {
             Some(parts.remove(0))
         };
+
         let version = if parts.is_empty() {
             None
         } else {
-            Some(parts.remove(0))
+            Some(parts.join("/"))
         };
+
         if let Some(mut ver) = version {
-            let underscore_index = ver.find('_');
             let mut peers_suffix = None;
-            if let Some(index) = underscore_index {
-                peers_suffix = Some(ver[index + 1..].to_string());
-                ver = ver[..index].to_string()
+
+            if ver.contains('(') && ver.ends_with(')') {
+                if let Some(index) = ver.find('(') {
+                    peers_suffix = Some(ver[index..].to_string());
+                    ver = ver[0..index].to_string();
+                }
+            } else {
+                if let Some(index) = ver.find('_') {
+                    peers_suffix = Some(ver[index + 1..].to_string());
+                    ver = ver[0..index].to_string();
+                }
             }
+
             if Version::parse(&ver).is_ok() {
-                return Ok(Self {
+                return PnpmDependencyPath {
                     host,
                     is_absolute,
                     name,
                     peers_suffix,
                     version: Some(ver),
-                });
+                };
             }
         }
-        if !is_absolute {
-            return Err(DependencyPathError::IsNotAbsolute(path.to_string()));
```

Comment:
I'm removing the error entirely. The error was copied from the JS implementation, but we really don't care if this is wrong, we simply need to proceed.